File: Python/Arrays/L_squareOfSortedArr.py
from typing import List


def sortedSquares(nums: List[int]) -> List[int]:
    l = 0
    r = len(nums)-1
    res = []
    while l<=r:
        if abs(nums[l])<abs(nums[r]):
            res.append(nums[r]**2)
            r -= 1
        else:
            res.append(nums[l]**2)
            l += 1
    return res[::-1]

File: cpp/Array/L_squareOfSortedArr.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> sortedSquares(vector<int> &nums)
{
    int n = nums.size(), l = 0, r = n - 1;
    vector<int> ans(n);
    for (int i = n - 1; i >= 0; i--)
    {
        int xl = nums[l] * nums[l], xr = nums[r] * nums[r];
        if (xl > xr)
        {
            ans[i] = xl;
            l++;
        }
        else
        {
            ans[i] = xr;
            r--;
        }
    }
    return ans;
}